src/faction_camp.cpp

```diff
@@ -1420,20 +1408,136 @@ void basecamp::craft_construction( npc &p, const std::string &cur_id, const std:
 {
-    std::string dir = camp_direction( task );
-    const tripoint omt_pos = p.global_omt_location();
-    tripoint omt_trg;
-    std::vector<std::pair<std::string, tripoint>> om_expansions = om_building_region( p, 1, true );
-    for( const auto &e : om_expansions ) {
-        if( dir == om_simple_dir( omt_pos, e.second ) ) {
-            omt_trg = e.second;
+    size_t harvestable = 0;
+    size_t plots_empty = 0;
+    size_t plots_plow = 0;
+    bool plow = op & farm_ops::plow;
+    bool plant = op & farm_ops::plant;
+    bool harvest = op & farm_ops::harvest;
+
+    const auto is_dirtmound = []( const tripoint & pos, tinymap & bay1, tinymap & bay2 ) {
+        const ter_str_id &dirtmound = ter_str_id( "t_dirtmound" );
+        const furn_str_id &f_null = furn_str_id( "f_null" );
+        return ( bay1.ter( pos ) == dirtmound ) && ( bay2.furn( pos ) == f_null );
+    };
+    const auto is_unplowed = []( const tripoint & pos, tinymap & bay ) {
+        return bay.ter( pos ) == ter_str_id( "t_dirt" ) || bay.ter( pos ) == ter_str_id( "t_grass" );
+    };
+    const auto is_valid_seed = []( const item & itm ) {
+        return itm.is_seed() && itm.typeId() != "marloss_seed" && itm.typeId() != "fungal_seeds";
+    };
+
+    std::vector<std::string> plant_names;
+
+    //bay_json is what the are should look like according to jsons
+    tinymap bay_json;
+    bay_json.generate( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, calendar::turn );
+    //bay is what the area actually looks like
+    tinymap bay;
+    bay.load( omt_tgt.x * 2, omt_tgt.y * 2, omt_tgt.z, false );
+    tripoint mapmin = tripoint( 0, 0, omt_tgt.z );
+    tripoint mapmax = tripoint( 2 * SEEX - 1, 2 * SEEY - 1, omt_tgt.z );
+    for( const tripoint &pos : bay.points_in_rectangle( mapmin, mapmax ) ) {
+        //Needs to be plowed to match json
+        if( is_dirtmound( pos, bay_json, bay ) && is_unplowed( pos, bay ) ) {
+            plots_plow += 1;
+        }
+        if( is_dirtmound( pos, bay, bay ) ) {
+            plots_empty++;
+        }
+        if( bay.furn( pos ) == furn_str_id( "f_plant_harvest" ) && !bay.i_at( pos ).empty() ) {
+            const item &seed = bay.i_at( pos ).front();
+            if( is_valid_seed( seed ) ) {
+                harvestable++;
+                const islot_seed &seed_data = *seed.type->seed;
+                item tmp = item( seed_data.fruit_id, calendar::turn );
+                if( std::find( plant_names.begin(), plant_names.end(),
+                               tmp.type_name( 3 ) ) != plant_names.end() ) {
+                    plant_names.push_back( tmp.type_name( 3 ) );
+                }
+            }
         }
     }
 
+    int total_c = 0;
+    for( const std::string &i : plant_names ) {
+        if( total_c < 5 ) {
+            crops += "\t" + i + " \n";
+            total_c++;
+        } else if( total_c == 5 ) {
+            crops += "+ more \n";
+            break;
+        }
+    }
+
+    if( plow ) {
```

Comment:
This is a bit confusing. The function handles all three cases at the same time, but only returns a value for one of the actions.

E.g. calling it as `camp_farm_act( pos, tmp, farm_ops::plow | farm_ops::plant | farm_ops::harvest )` seems to be possible and is actually handled, but what is returned: the number of tiles to plow, to plant or to harvest?

I suggest changing
```C++
bool plow = op & farm_ops::plow;
// into
bool plow = op == farm_ops::plow;
```
And add a `debugmsg` when none of the booleans is `true`. Probably the same with the functions below.